src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -285,11 +285,14 @@ export function makeHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sam
                             trackGroup.genes.delete(gene);
                             if (!trackGroup.genes.size) {
                                 molecularProfileIdToHeatmapTracks.delete(molecularProfileId);
-                                if (oncoprint.sortMode.type === "heatmap" && oncoprint.sortMode.clusteredHeatmapProfile === molecularProfileId) {
```

Comment:
why move this down?

Reply:
This removes duplicate logic to say ‘reset sorting when all tracks in the clustered profile are removed’.

When the tracks have been removed because the ‘Remove heatmap’ button has cleared the map via the code you commented on [above](https://github.com/cBioPortal/cbioportal-frontend/pull/982#discussion_r170300827), the callback does not need to update the map, but it does have to reset sorting.